request-manager/grapher/grapher.go

```diff
@@ -692,6 +695,12 @@ func (o *Grapher) newNoopNode(name string, nodeArgs map[string]interface{}) (*No
 
 // newNode creates a node for the given job j
 func (o *Grapher) newNode(j *NodeSpec, nodeArgs map[string]interface{}) (*Node, error) {
+	// Make a copy of the nodeArgs before this node gets created and potentially
+	// adds additional keys to the nodeArgs.
+	originalArgs := map[string]interface{}{}
+	for k, v := range nodeArgs {
+		originalArgs[k] = v
+	}
```

Comment:
It should not be possible for values in the nodeArgs map to change, but you bring up a good point so I will add a comment explaining that. Good eye Felix! There are a few other spots in the grapher where we copy args maps as well